src/main/java/core/basesyntax/handler/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.handler;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.FruitStorage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final FruitStorage fruitStorage;
+
+    public PurchaseOperationHandler(FruitStorage fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        fruitStorage.removeFruit(transaction.getFruit(), transaction.getQuantity());
+    }
```

Comment:
According to the checklist, for the Purchase operation, you need to check the result balance before saving it in the Storage to ensure it is positive. If the balance is negative, a RuntimeException should be thrown. Currently, there is no such check implemented in the `apply` method.